packages/react-api/src/Api.tsx

```diff
@@ -212,14 +205,49 @@ function Api ({ children, store, url }: Props): React.ReactElement<Props> | null
         .then(setState)
         .catch((error): void => {
           console.error(error);
-
           setApiError((error as Error).message);
         });
-    });
+    }
+
+    async function substrateConnectApi (network: string) {
+      const detect = new Detector('PolkadotJS apps');
+      const signer = new ApiSigner(registry, queuePayload, queueSetTxStatus);
+      const options = { registry, signer, types, typesBundle, typesChain };
+
+      try {
+        api = await detect.connect(network, undefined, options);
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        api.isReady.then(() => {
+          setIsApiInitialized(true);
+          setIsApiConnected(true);
+          actWhenReady(types);
+        }).catch(console.error);
+      } catch (err) {
+        console.error(err);
+      }
+    }
+
+    const types = getDevTypes();
 
-    setIsApiInitialized(true);
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    if (apiType?.type === 'substrate-connect') {
+      // eslint-disable-next-line no-void
+      substrateConnectApi(apiType.param).catch(console.error);
+    } else if (apiType?.type === 'json-rpc') {
```

Comment:
This is a bit of an issue - since `apiType` is optional, we actually could end up where we do nothing now. contrast this to the original where it always constructed an API instance.

So only check for light, if not, always follow the URL path.

Reply:
I would actually make it mandatory, as few lines below the WSProvider cannot be created without it (and type is not optional in apiType):
```
const provider = new WsProvider(apiType.param);
```
Afair apiType should always be provided from now on in the Endpoint type format

Reply:
Happy to have it mandator and remove all the `apiType?.` checks everywhere.